libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1074,25 +1074,39 @@ bool TypeChecker::visit(Conditional const& _conditional)
 
 	TypePointer trueType = type(_conditional.trueExpression())->mobileType();
 	TypePointer falseType = type(_conditional.falseExpression())->mobileType();
+
+	TypePointer commonType;
+
 	if (!trueType)
-		m_errorReporter.fatalTypeError(_conditional.trueExpression().location(), "Invalid mobile type.");
+		m_errorReporter.typeError(_conditional.trueExpression().location(), "Invalid mobile type in true expression.");
+	else
+		commonType = trueType;
+
 	if (!falseType)
-		m_errorReporter.fatalTypeError(_conditional.falseExpression().location(), "Invalid mobile type.");
+		m_errorReporter.typeError(_conditional.falseExpression().location(), "Invalid mobile type in false expression.");
+	else
+		commonType = falseType;
 
-	TypePointer commonType = Type::commonType(trueType, falseType);
-	if (!commonType)
+	if (!trueType && !falseType)
+		BOOST_THROW_EXCEPTION(FatalError());
```

Comment:
Is it ok to explicitly throw the exception instead of doing it via the error reporter?

Reply:
I found no other way to cause a fatal error without causing a new error message

Reply:
I think this is also done in other cases.